src/v2/cookbook/form-validation.md

```diff
@@ -0,0 +1,322 @@
+
+<p data-height="265" data-theme-id="0" data-slug-hash="vWqNXZ" data-default-tab="html,result" data-user="cfjedimaster" data-embed-version="2" data-pen-title="form validation 2" class="codepen">See the Pen <a href="https://codepen.io/cfjedimaster/pen/vWqNXZ/">form validation 2</a> by Raymond Camden (<a href="https://codepen.io/cfjedimaster">@cfjedimaster</a>) on <a href="https://codepen.io">CodePen</a>.</p>
+<script async src="https://production-assets.codepen.io/assets/embed/ei.js"></script>
+
+## Another Example of Custom Validation
+
+For the third example, we've built something you've probably seen in survey apps. The user is asked to spend a "budget" for a set of features for a new Star Destroyer model. The total must equal 100. First, the HTML.
+
+``` html
+<form id="app" @submit="checkForm" action="/something" method="post" novalidate="true">
+  
+  <p v-if="errors.length">
+    <b>Please correct the following error(s):</b>
+    <ul>
+      <li v-for="error in errors">{{ error }}</li>
+    </ul>
+  </p>
+
+  <p>
+    Given a budget of 100 dollars, indicate how much
+    you would spend on the following features for the
+    next generation Star Destroyer:
+  </p>
+
+  <p>
+    <input type="number" name="weapons" v-model.number="weapons"> Weapons <br/>
+    <input type="number" name="shields" v-model.number="shields"> Shields <br/>
+    <input type="number" name="coffee" v-model.number="coffee"> Coffee <br/>
+    <input type="number" name="ac" v-model.number="ac"> Air Conditioning <br/>
+    <input type="number" name="mousedroids" v-model.number="mousedroids"> Mouse Droids <br/>
+  </p>
+
+  <p>
+    Current Total: {{total}}
+  </p>
+
+  <p>
+    <input type="submit" value="Submit">  
+  </p>
+
+</form>
+```
+
+Note the set of inputs covering the five different features. Note the addition of .number to the v-model attribute. This tells Vue to cast the value to a number when you use it. However, there is a bug with this feature such that when the value is blank, it turns back into a string. You'll see the workaround below. To make it a bit easier for the user, we also added a current total right below so they can see, in real time, what their total is. Now let's look at the JavaScript.
+
+``` js
+const app = new Vue({
+  el:'#app',
+  data:{
+    errors:[],
+    weapons:0,
+    shields:0,
+    coffee:0,
+    ac:0,
+    mousedroids:0
+  },
+  computed:{
+     total:function() {
+       //must parse cuz Vue turns empty value to string
+       return Number(this.weapons)+
+         Number(this.shields)+
+         Number(this.coffee)+
+         Number(this.ac+this.mousedroids);
+     }
+  },
+  methods:{
+    checkForm:function(e) {
+      this.errors = [];
+      if(this.total != 100) this.errors.push("Total must be 100!");
+      if(!this.errors.length) return true;
+      e.preventDefault();
+    }
+  }
+})
+```

```

Comment:
This seems like it needs to be kept from going below 0. The negative numbers don't make much sense if the user clicks down instead of up

Reply:
One could see it as a strong vote against. :)

Reply:
This goes for the first example with the age (or maybe built into the second one) as well. Thanks!

Reply:
@cfjedimaster good point! Maybe we leave this example as is but change the first one with the age. Pretty sure you can't be negative years old :)

Reply:
I've added `min="0"` to the article (not yet in code), please add to the CodePen. Here is the complete line for reference:

    <input type="number" name="age" id="age" v-model="age" min="0">